lib/spack/spack/test/config.py

```diff
@@ -1266,3 +1266,63 @@ def test_user_cache_path_is_default_when_env_var_is_empty(working_env):
     os.environ['SPACK_USER_CACHE_PATH'] = ''
     assert os.path.expanduser("~%s.spack" % os.sep) == \
         spack.paths._get_user_cache_path()
+
+
+github_url = "https://github.com/fake/fake/{0}/develop"
+gitlab_url = "https://gitlab.fake.io/user/repo/-/blob/config/defaults"
+
+
+@pytest.mark.parametrize('url,isfile,fail', [
+    ("https://not.a.real.url/", False, True),
+    (github_url.format('tree'), False, True),
+    ("{0}/etc/fake/defaults".format(github_url.format('tree')), False, False),
+    ("{0}/README.md".format(github_url.format('blob')), True, True),
+    ("{0}/etc/fake/defaults/packages.yaml".format(github_url.format('blob')),
+     True, False),
+    (gitlab_url, False, False),
+    ("{0}/compilers.yaml".format(gitlab_url), True, False),
+    (None, False, True)
+])
+def test_config_fetch_remote_configs(
+        tmpdir, mutable_empty_config, mock_url_config, url, isfile, fail):
+    dest_dir = join_path(tmpdir.strpath, 'defaults')
+    if fail:
+        msg = "Cannot retrieve configuration"
+        with spack.config.override('config:url_fetch_method', 'curl'):
+            with pytest.raises(spack.config.ConfigFileError, match=msg):
+                spack.config.fetch_remote_configs(url, dest_dir)
+    else:
+        with spack.config.override('config:url_fetch_method', 'curl'):
+            path = spack.config.fetch_remote_configs(url, dest_dir)
+            assert path
+            assert os.path.exists(path)
+            if isfile:
+                assert os.path.basename(path) == os.path.basename(url)
+
+
+def test_config_file_dir_failure(tmpdir, mutable_empty_config):
+    with pytest.raises(spack.config.ConfigFileError, match="not a file"):
+        spack.config.read_config_file(tmpdir.strpath)
```

Comment:
(no change needed) This seems fine although it is somewhat generic (not strictly related to the PR goals) - I don't see any similar tests.

Reply:
This was for coverage and tests a `read_config_file` path that was not previously covered.